### DISCUSSION TITLE
My own color not working

### DISCUSSION THREAD

**Discussion by techinp** [heroui-inc/heroui#544] [Help] [answered]
I have followed every step on [Document](https://nextui.org/docs/theme/customize-theme#customizing-theme-tokens) to add new color But it doesn't work.

Here is my code:

pages/_app.js
```
import { createTheme, NextUIProvider } from '@nextui-org/react';

const theme = createTheme({
  type: 'light',
  theme: {
    colors: {
      myColor: '#ff4ecd',
      // ...  more colors
    },
    space: {},
    fonts: {},
  },
});

function App({ Component, pageProps }) {
  return (
    <NextUIProvider theme={theme}>
      <Component {...pageProps} />
    </NextUIProvider>
  )
}

export default App
```

component/header.js
```
import { Text } from '@nextui-org/react';
export default function Header({}) {
  return (
    <>
      <Text css={{ color: '$myColor'}}>Foo</Text>
    </>
  )
}
```

Here is result from inspect styles
<img width="534" alt="Screen Shot 2565-06-16 at 03 53 26" src="https://user-images.githubusercontent.com/44148263/173928664-a6f38137-d1a3-4869-8905-897358580687.png">
<img width="530" alt="Screen Shot 2565-06-16 at 03 53 02" src="https://user-images.githubusercontent.com/44148263/173928697-c440d902-baa4-43eb-b4af-fca8b378e88b.png">

**Comment by tianenpang** [comment]
Hi @techinp I can' t reproduce it ([codesandbox example](https://codesandbox.io/s/nextui-discussion-544-6ns9he?file=/pages/index.tsx)), but it looks like a mistake in css variable generation. 

Have you tried restart the dev server?

**Comment by techinp** [comment]
Thank you for your example and suggestion.

I think I have to find out what is something wrong in my project.

**Comment by techinp** [comment, marked as answer]
I found the thing that make my project can't use custom theme

if your project use [styled-component](https://styled-components.com/) and set `ssr: true` in `.babelrc` file the new css in custom theme will not working

here is my code
`.babelrc`
```
{
  "presets": ["next/babel"]
  "plugins": [["styled-components", { "ssr": true }]] // this line
}

```

**Comment by tianenpang** [comment]
Since Next.js 12, the swc compiler supports styled-component, you can use this feature instead of babel and works well with NextUI. 

[codesandbox example (updated)](https://codesandbox.io/s/nextui-discussion-544-6ns9he?file=/pages/index.tsx)

Configure in `next.config.js`

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  // others...
  compiler: {
    styledComponents: true
  }
}

module.exports = nextConfig
```

[nextjs support docs](https://nextjs.org/docs/advanced-features/compiler#styled-components)
[styled-components docs](https://styled-components.com/docs/advanced#with-swc)
[nextjs with styled-components examples](https://github.com/vercel/next.js/tree/canary/examples/with-styled-components)
